internal/convert.go

```diff
@@ -29,26 +29,29 @@ import (
-	Audit          Audit               `json:"-"` // Stores the audit information for the database conversion
-	Rules          []Rule              // Stores applied rules during schema conversion
-	IsSharded      bool                // Flag denoting if the migration is sharded or not
-	ConvLock       sync.RWMutex        `json:"-"` // ConvLock prevents concurrent map read/write operations. This lock will be used in all the APIs that either read or write elements to the conv object.
+	mode               mode                     // Schema mode or data mode.
+	SpSchema           ddl.Schema               // Maps Spanner table name to Spanner schema.
+	SyntheticPKeys     map[string]SyntheticPKey // Maps Spanner table name to synthetic primary key (if needed).
+	SrcSchema          map[string]schema.Table  // Maps source-DB table name to schema information.
+	SchemaIssues       map[string]TableIssues   // Maps source-DB table/col to list of schema conversion issues.
+	ToSpanner          map[string]NameAndCols   `json:"-"` // Maps from source-DB table name to Spanner name and column mapping.
+	ToSource           map[string]NameAndCols   `json:"-"` // Maps from Spanner table name to source-DB table name and column mapping.
+	UsedNames          map[string]bool          `json:"-"` // Map storing the names that are already assigned to tables, indices or foreign key contraints.
+	dataSink           func(table string, cols []string, values []interface{})
+	DataFlush          func()              `json:"-"` // Data flush is used to flush out remaining writes and wait for them to complete.
+	Location           *time.Location      // Timezone (for timestamp conversion).
+	sampleBadRows      rowSamples          // Rows that generated errors during conversion.
+	Stats              stats               `json:"-"`
+	TimezoneOffset     string              // Timezone offset for timestamp conversion.
+	SpDialect          string              // The dialect of the spanner database to which Spanner migration tool is writing.
+	UniquePKey         map[string][]string // Maps Spanner table name to unique column name being used as primary key (if needed).
+	Audit              Audit               `json:"-"` // Stores the audit information for the database conversion
+	Rules              []Rule              // Stores applied rules during schema conversion
+	IsSharded          bool                // Flag denoting if the migration is sharded or not
+	ConvLock           sync.RWMutex        `json:"-"` // ConvLock prevents concurrent map read/write operations. This lock will be used in all the APIs that either read or write elements to the conv object.
+	SpRegion           string              // Region for Spanner Instance
+	ResourceValidation bool                // Flag denoting if validation for resources to generated is complete
+	UI                 bool                // Flag if UI interface was used for migration. ToDo: Remove flag after resource generation is introduced to UI
```

Comment:
Where do we set this?

Reply:
My bad, its in the UI bit. I thought i would add the changes as part of my UI Pr, but you're right. It needs to go with this PR